lib/io/src/i2c.rs

```diff
@@ -21,4 +21,11 @@ pub trait HypedI2c {
         register_address: u8,
         data: u8,
     ) -> Result<(), I2cError>;
+    fn read_byte_16(&mut self, device_address: u8, register_address: u16) -> Option<u8>;
```

Comment:
The C++ code only required 16-bit register addresses for the `writeByteToRegister` function, is it needed for `read_byte` too?

Reply:
good point; `read_byte_16` isn't needed and will be removed